kubectl-plugin/pkg/cmd/scale/scale.go

```diff
@@ -0,0 +1,150 @@
+package scale
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/completion"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type ScaleClusterOptions struct {
+	configFlags *genericclioptions.ConfigFlags
+	ioStreams   *genericclioptions.IOStreams
+	replicas    *int32
+	workerGroup string
+	cluster     string
+}
+
+var (
+	scaleLong = templates.LongDesc(`
+		Scale a Ray cluster's worker group.
+	`)
+
+	scaleExample = templates.Examples(`
+		# Scale a Ray cluster's worker group to 3 replicas
+		kubectl ray scale my-workergroup --raycluster my-raycluster --replicas 3
+	`)
+)
+
+func NewScaleClusterOptions(streams genericclioptions.IOStreams) *ScaleClusterOptions {
+	return &ScaleClusterOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+		ioStreams:   &streams,
+		replicas:    new(int32),
+	}
+}
+
+func NewScaleClusterCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewScaleClusterOptions(streams)
+	// Initialize the factory for later use with the current config flag
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:               "scale [WORKERGROUP] [-c/--raycluster CLUSTERNAME] [-r/--replicas N]",
+		Short:             "Scale Ray cluster",
+		Long:              scaleLong,
+		Example:           scaleExample,
+		SilenceUsage:      true,
+		ValidArgsFunction: completion.RayClusterResourceNameCompletionFunc(cmdFactory),
```

Comment:
If we decide to make both cluster name and workergroup name as required arguments, we need to write a new shell completion function here.

Reply:
removed this line. Will leave for future PR.